### PR TITLE
Ian/validation errors

### PR DESCRIPTION
**Pull request by ianwremmel**
This is based on @ssuttner's work in #31

### CODE REVIEW

File: src/components/press-daterangepicker/press-daterangepicker.vue [hunk 1]
```diff
@@ -1,10 +1,18 @@
...
+      v-once
+      :name="name"
+      :value="localeStringeFromValue"
+      data-press-daterangepicker-input
+    >
+    <div
+      v-if="pressValidationError"
+      class="ui error input"
+    >
+      {{ pressValidationError }}
```
Comment (ssuttner) [inline src/components/press-daterangepicker/press-daterangepicker.vue]: I think it is OK to be able to pass in as a prop with the validation error text that we want to use.
But, whether the validation error is displayed or not, should be dynamic based on an error class being there or not.
That's why I tried to do here: https://github.com/UrbanDoor/press/pull/31/commits/36a1706650b4e316fcd34f9344b8875d5c97206b#diff-a81c8385c8a76db71073c1a5b8ea746cR67

File: src/components/press-datepicker/press-datepicker.vue [hunk 2]
```diff
@@ -20,6 +30,14 @@ export default {
       default: '',
       type: String
     },
+    pressValidationError: {
+      default: '',
+      type: String
+    },
+    required: {
```
Comment (ssuttner) [inline src/components/press-datepicker/press-datepicker.vue]: the required `*` label class will go in a separate PR, right?

File: src/components/press-daterangepicker/press-daterangepicker.vue (hunk 1 above)
Comment (ianwremmel) [reply src/components/press-daterangepicker/press-daterangepicker.vue]: Shouldn't this be just as effective? We'll need to tweak the wrapper a bit, but shouldn't we have access to that message (in ruby land) any time the server would add an error class?

File: src/components/press-datepicker/press-datepicker.vue (hunk 2 above)
Comment (ianwremmel) [reply src/components/press-datepicker/press-datepicker.vue]: Take a look at the html in `validation.ejs`. I wrote this in a way that'll require a bit more change to the rails form stuff, but should keep some of the magic out of this code.

File: src/components/press-daterangepicker/press-daterangepicker.vue (hunk 1 above)
Comment (ssuttner) [reply src/components/press-daterangepicker/press-daterangepicker.vue]: OK, yes, if we tweak the wrapper it might work.
I was just thinking in terms of the current simple_form behavior... Let's try it.

File: src/components/press-datepicker/press-datepicker.vue (hunk 2 above)
Comment (ssuttner) [reply src/components/press-datepicker/press-datepicker.vue]: OK, I'm curious to see what you have in mind there.
I'm ok with the label support not being part of this PR or this specific code.
